### ISSUE TITLE
Findable help (3.2.6) demands contact links within PDF's for instance?

### ISSUE THREAD

**Issue by ShadowBB** [w3c/wcag#1693]
Currently this SC demands that all pages have at least 1 option to access the information in the same relative order.

1. I think this needs to be "all pages within the set of web pages".
2. The current definition of "set of web pages" and "web page" causes for example PDF documents to be counted among web pages within that set of web pages (because they are most likely made by the same organization with the same purpose as the website).

That means that every PDF now for example needs a link to the contact page. I don't think that is what this SC is trying to accomplish.

**Comment by alastc** [comment]
Hi @ShadowBB,

Do you mean the [current version](https://w3c.github.io/wcag/guidelines/22/#consistent-help)?

> For each web page within a set of web pages that provides one or more of following ways of finding help, access to at least one form of help is included in the same relative order on each page:
> - Human contact details;
> - Human contact mechanism;
> - Self-help option;
> - A fully automated contact mechanism.

I hadn't interpreted a web page in a set-of to include PDFs. For example, they don't include the same navigation. 

We can call that out in the understanding doc if that helps?

**Comment by ShadowBB** [comment]
A set of web pages is defined as "collection of web pages that share a common purpose and that are created by the same author, group or organization ". A PDF would not share the same navigation but would still be part of the same set of web pages if they are created by the same author, group or organization. So a PDF is definitely part of the same set of web pages.

However, upon reading this SC again there could be 2 ways of interpreting it. 

1. For each web page (within a set of web pages that provides one or more of following ways of finding help), access to at least one form of help is included in the same relative order on each page:
2. For each (web page within a set of web pages) that provides one or more of following ways of finding help, access to at least one form of help is included in the same relative order on each page:

Does the "web page" or the "set of web pages" need to provide "one or more of following ways of finding help" for this SC to be relevant?

In the first case the PDF only needs to have that option in the same relative order if it actually has one of those 4 forms of help included.

In the second case the PDF needs to have that option period.

I think both are unwanted.

And PDF's are just one example. What if you have a completely different system that doesn't share the same header/footer template but would still be part of the same set of web pages?

For example. See the difference between these two pages that are clearly part of the same set of web pages:
https://www.rotterdam.nl/
https://sso.ir.rotterdam.nl/mijnloket/

They are part of the same set of web pages but the second one doesn't have a link to the contact page.

Should this second page pass because according to this SC it doesn't have any form of help at all (despite the "help" link) or should it fail because it doesn't have one of the 4 options in the same relative order?

**Comment by mraccess77** [comment]
@ShadowBB I think that is the question - I think the language is clear that if the set doesn't contain help then you don't have to provide it.  But, if a page in the set has the help but not all pages in the set have the help does this wording mean that it would have to be provided for those other pages.

**Comment by alastc** [comment]
> Does the "web page" or the "set of web pages" need to provide "one or more of following ways of finding help" for this SC to be relevant?

The intent is to match the consistent navigation language, so that the requirement is having something in the same location (order) if you have it, but not to fail if you don't have it. That ordering requirement is constrained to the set of pages.


> For example. See the difference between these two pages that are clearly part of the same set of web pages:

Would you fail those on consistent navigation? The main nav is earlier on the page in the second page.
The contact link also disappears from the second page, so it's either failing both or neither.

Regarding PDFs specifically, I've never considered those as part of a 'set of web pages' because they don't share the same features as the (HTML) web pages they are linked from. I don't have a more concrete reference for that off-hand, but I seem to remember @bruce-usab was vocal about this, perhaps he remembers?

**Comment by mraccess77** [comment]
@alastc I think the question is what happens if 3 out of 4 pages in a set have the help in a consistent location.  Does that one page fail or does it pass because 
not doesn't have the help option.

**Comment by ShadowBB** [comment]
> 
> 
> > Does the "web page" or the "set of web pages" need to provide "one or more of following ways of finding help" for this SC to be relevant?
> 
> The intent is to match the consistent navigation language, so that the requirement is having something in the same location (order) if you have it, but not to fail if you don't have it. That ordering requirement is constrained to the set of pages.

If the intent is to match the consistent navigation language I think we should follow it more closely. Is this SC meant as a subset of "3.2.3 consistent navigation" but on a single A level?

> 
> > For example. See the difference between these two pages that are clearly part of the same set of web pages:
> 
> Would you fail those on consistent navigation? The main nav is earlier on the page in the second page.
> The contact link also disappears from the second page, so it's either failing both or neither.

I would not fail those pages on consistent navigation because the "Navigational mechanisms that are repeated" have the same relative order. There are no other items that where earlier in the one page and later than those navigational mechanisms on the other page nor vice versa.

I would fail those pages on 3.2.6 because it's a web page (within a set of web pages that provides help) but none of the 4 options is provided on the "mijnloket" page and the SC clearly says that at least one of those 4 options should be provided.

> 
> Regarding PDFs specifically, I've never considered those as part of a 'set of web pages' because they don't share the same features as the (HTML) web pages they are linked from. I don't have a more concrete reference for that off-hand, but I seem to remember @bruce-usab was vocal about this, perhaps he remembers?

They usually don't share the same features. I agree with that. But the definition says they should "share a common purpose". But now that I reread the example I think up until now I have been using a to broad definition of "purpose". It's a really ambiguous definition.

**Comment by alastc** [comment]
> I would fail those pages on 3.2.6 because it's a web page (within a set of web pages that provides help) but none of the 4 options is provided on the "mijnloket" page and the SC clearly says that at least one of those 4 options should be provided. 

I wouldn't, because the second doesn't provide the mechanism of help (contact) at all. It needs to be present on the page, and (at least one) other page in the set.

The answers should be:
- pass if you have the help in the same order.
- pass if you don't have the help at all.
- Fail if you have the help, but not in the same order.

Would putting commas around that help? 

> For each web page, within a set of web pages, that provides one or more of following ways of finding help, access to at least one form of help is included in the same relative order on each page:

I tried putting the 'set of web pages' at the end, but you could still read it either way :-/

**Comment by ShadowBB** [comment]
You are correct that putting "set of web pages" at the end would not help this problem. That was a second (minor) issue that I should have probably made a different github issue about.

Putting in commas might help a bit, but strictly speaking it could still be interpreted two ways. Its just a quirk of human English language I suppose. Specifying it might work:
 
`For each web page within a set of web pages that provides one or more of following ways of finding help somewhere within that set, access to at least one form of help is included in the same relative order on each page within the set of web pages if it is included at all on that page.`

I think that covers both of the issues.

**Comment by mraccess77** [comment]
@alastc If you have help on 2 pages in a set in relative order but not help on 1 page in the same set is that a pass as well?

**Comment by alastc** [comment]
@mraccess77 Yes. Not ideal, but prevents false-positives.

**Comment by alastc** [comment]
Just coming back to this issue as I realised we hadn't finished off the survey answer.

I think it has been overtaken by other changes, the current text is:
> If a [web page](https://w3c.github.io/wcag/guidelines/22/#dfn-web-page-s) contains any of the following help mechanisms, and those mechanisms are repeated on multiple web pages within a [set of web pages](https://w3c.github.io/wcag/guidelines/22/#dfn-set-of-web-pages), they occur in the same relative order to other page content, unless a change is initiated by the user:

So I think that lines up with [interpretation 1 from above](https://github.com/w3c/wcag/issues/1693#issuecomment-805322776):
> For each web page (within a set of web pages that provides one or more of following ways of finding help), access to at least one form of help is included in the same relative order on each page:

As that was the desired outcome I'll close this issue, but please re-open if I've misunderstood.

**Comment by GreggVan** [comment]
Good but you missed one part.

"Same relative order — related to what?"


To fix that you need to add back "to other page content"


For each web page (within a set of web pages that provides one or more of following ways of finding help), access to at least one form of help is included in the same relative order to other page content on each page:



Gregg Vanderheiden
***@***.***
